lib/core/viewer/components/pdfViewer.component.ts

```diff
@@ -436,13 +438,14 @@ export class PdfViewerComponent implements OnChanges, OnDestroy {
         this.dialog
             .open(PdfPasswordDialogComponent, {
                 width: '400px',
-                disableClose: true,
                 data: { reason }
             })
             .afterClosed().subscribe((password) => {
-            if (password) {
-                callback(password);
-            }
+                if (password) {
+                    callback(password);
+                } else {
+                    this.location.back();
```

Comment:
@davidcanonieto has introduced a new service to go back in a better way can we use it https://github.com/Alfresco/alfresco-ng2-components/blob/development/lib/core/viewer/components/viewer.component.ts   ?  private previousRouteService: PreviousRouteService)

Reply:
doesn't seem to work.

Reply:
@davidcanonieto can you please check with @pionnegru if is possible to reuse your service or it needs some improvement?